core/src/main/java/io/lionweb/lioncore/java/serialization/PrimitiveValuesSerialization.java

```diff
@@ -72,6 +72,29 @@ public PrimitiveValuesSerialization registerSerializer(
     return this;
   }
 
+  private StructuredDataTypeInstance deserializeSDT(String dataTypeID, JsonObject jo) {
+    StructuredDataType sdt = strucuturesDataTypesByID.get(dataTypeID);
+    DynamicStructuredDataTypeInstance sdtInstance = new DynamicStructuredDataTypeInstance(sdt);
+    for (Field field : sdt.getFields()) {
+      if (jo.has(field.getName())) {
```

Comment:
According to spec, we use the field's key in the JSON structure, not it's name.

Reply:
Right, it should be now fixed